src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1496,7 +1496,7 @@ private BoundExpression BindNonMethod(SimpleNameSyntax node, Symbol symbol, Diag
 
                 case SymbolKind.Alias:
                     {
-                        var alias = symbol as AliasSymbol;
+                        var alias = (AliasSymbol)symbol;
                         symbol = alias.Target;
```

Comment:
I don't know if it's forbidden but the pattern being replaced certainly seems sub-optimal.  It would make investigating crash dumps significantly harder than the new shape of the code.